README.md

```diff
@@ -1,2 +1,187 @@
-# js-signer
- This repository provides library to sign any arbitrary message with an EOSIO R1 key and produce an EOSIO signature
+# NodeJS/Javascript library for Signing Messages
+
+This repository provides library to sign any arbitrary message with an EOSIO R1 key and produce an EOSIO signature
+
+## How to use in NodeJs
+
+### Online links for NodeJs
+
+- [Code / Playground Link 🔗](https://stackblitz.com/edit/node-js-signer?file=index.js)
+
+```js
+// run `node index.js` in the terminal
+
+// import the library
+const getSignature = require('js-signer').getSignature
+
+// import public private keys
+// for demo: keys are hardcoded here
+const privateKeyString = 'PVT_R1_ENSnpAGb4NHNA2chipxHQMVnAZdEAfRzHmJFEuxFkWvCXC5CG'
+const publicKeyString = 'PUB_R1_8CHJquaQWe4Pkhp1fBR9deP5wkqfjuWdfhaKYDxGKCo7gQwU9C'
+
+// create a function to generate singature for payload
+// this functions internally calls getSignature from library
+const signMessage = async message => await getSignature(message, publicKeyString, privateKeyString)
+
+// call the function with payload to generate signature
+const sigature = signMessage({ message: 'A very secret message' })
+
+// use the signature to add in the header of the API communication
+// for demo: print the signature
+console.log(signature)
+```
+
+## How to use in react
+
+### Online links for React
+
+- [Demo Link 🔗](https://react-ts-9q3wkh.stackblitz.io)
+- [Code / Playground Link 🔗](https://stackblitz.com/edit/react-ts-9q3wkh?file=App.tsx)
```

Comment:
The design is very funky, maybe a bit too funky for something we'd send to our clients. 
I think it is better in this case to bring it in line with our exchange design.